lib/ansible/module_utils/connection.py

```diff
@@ -146,21 +146,26 @@ def _exec_jsonrpc(self, name, *args, **kwargs):
             data = req.get('params')
 
             if isinstance(data, dict):
-                data = data.get('var_options', {})
-
-                for key, value in iteritems(data):
-                    try:
-                        dummy = json.dumps(value)
-                    except TypeError:
-                        raise ConnectionError(
-                            "Failed to encode some variables as JSON for communication with ansible-connection. "
-                            "Please open an issue and mention that the culprit is most likely '%s'" % key
-                        )
-
-            raise ConnectionError(
-                "Failed to encode some variables as JSON for communication with ansible-connection. "
-                "The original exception was: %s" % to_text(exc)
-            )
+                options = data.get('var_options', {})
+
+                for key, value in iteritems(options):
```

Comment:
This works when he vault is on the top level, but vaults can appear at any point int the data structure:
```yaml
nonvault:
   key1: stillnovaul
   key2:
       otherdict: otherval
       thisisvault: !vault ...
```